tsdb/engine/tsm1/engine.go

```diff
@@ -120,7 +122,15 @@ func (e *DevEngine) WritePoints(points []models.Point, measurementFieldsToSave m
 		}
 	}
 
-	return e.WAL.WritePoints(values)
+	id, err := e.WAL.WritePoints(values)
+
+	// Write data to cache for query purposes.
+	for k, v := range values {
```

Comment:
Since every call to write grabs a lock, it would be better to have a method on the cache that can take the collection of values so that we only lock once.